lib/src/model/discovered_device.dart

```diff
@@ -27,13 +27,19 @@ class DiscoveredDevice extends $DiscoveredDevice {
   @CustomEquality(DeepCollectionEquality())
   final Map<Uuid, Uint8List> serviceData;
 
+  /// Returns a map of Company Identifier Codes and the corresponding manufacturer
+  /// specific data.
+  @CustomEquality(DeepCollectionEquality())
```

Comment:
@werediver actually here we are touching an interesting point. Do we consider a device with different servicedata and / or different manufacturerdata but with same id (macadress on android, uuid iOS) a new device or the same.  I think I can advocate for both cases

Reply:
@remonh87 I do not agree with the current implementation at all. We should not abuse the equality test operator to perform some particular "matching", as in ID comparison or approximate comparison or anything. We shouldn't ignore fields at all (we do now).